test/lexer.jl

```diff
@@ -407,6 +408,8 @@ end
     @test tok("2048f0").kind == Tokens.FLOAT
     @test tok("1.:0").kind == Tokens.FLOAT
     @test tok("1.?").kind == Tokens.FLOAT
+    @test tok("0x00p2").kind == Tokens.FLOAT
+    @test tok("0x00P2").kind == Tokens.FLOAT
```

Comment:
More tests (I think they fail at the moment):

```
0x0.00p23
0x0.0_0p2
0x0_0_0.0_0p2
```

Reply:
added,  the numbers trailing the `.` can be hex right?

Reply:
Yes

```
julia> 0x0_0_0.ffp22e
ERROR: syntax: invalid numeric constant "0x000.ffp22e"

julia> 0x0_0_0.ffp22f
4.17792e6
```

Why does the `f` work?

Reply:
looks like a mistake to me, f0 also works